src/access/AccessControl.h

```diff
@@ -353,12 +354,35 @@ class AccessControl
 
     AccessControl() = default;
 
-    AccessControl(Delegate & delegate) : mDelegate(delegate) {}
-
     AccessControl(const AccessControl &) = delete;
     AccessControl & operator=(const AccessControl &) = delete;
 
-    ~AccessControl() { mDelegate.Release(); }
+    ~AccessControl() { mDelegate->Release(); }
+
+    /**
+     * @brief Set the delegate that will handle access control entries. Must be called before Init().
+     *
+     * @param delegate - The delegate to use
+     * @return CHIP_NO_ERROR on success, CHIP_ERROR_INCORRECT_STATE if set after Init(), and
+     *         CHIP_ERROR_INVALID_ARGUMENT if delegate is null.
+     */
+    CHIP_ERROR SetDelegate(Delegate * delegate)
+    {
+        if (mIsInitialized)
+        {
+            return CHIP_ERROR_INCORRECT_STATE;
+        }
+        else if (delegate == nullptr)
+        {
+            return CHIP_ERROR_INVALID_ARGUMENT;
+        }
+
+        // De-initialize the default-set delegate.
```

Comment:
This may not only be the default-set in case I somehow call SetDelegate before. Maybe comment could say that "any previously set" instead and give the default as an example. If we only want to ever override once the default, we could add a check and only allow SetDelegate to be called once (i.e. only when mDelegate is default)

Reply:
Removed